http.go

```diff
@@ -161,3 +155,96 @@ func startHTTPHandler(state *State) {
 	}
 	log.Fatal(r.Run(*httpAddr))
 }
+
+// Round duration to N decimal points. Original source here: https://play.golang.org/p/WjfKwhhjL5
+func RoundN(d time.Duration, n int) time.Duration {
+	if n < 1 {
+		return d
+	}
+	if d >= time.Hour {
+		k := digits(d / time.Hour)
+		if k >= n {
+			return d.Round(time.Hour * time.Duration(math.Pow10(k-n)))
+		}
+		n -= k
+		k = digits(d % time.Hour / time.Minute)
+		if k >= n {
+			return d.Round(time.Minute * time.Duration(math.Pow10(k-n)))
+		}
+		return d.Round(time.Duration(float64(100*time.Second) * math.Pow10(k-n)))
+	}
+	if d >= time.Minute {
+		k := digits(d / time.Minute)
+		if k >= n {
+			return d.Round(time.Minute * time.Duration(math.Pow10(k-n)))
+		}
+		return d.Round(time.Duration(float64(100*time.Second) * math.Pow10(k-n)))
+	}
+	if k := digits(d); k > n {
+		return d.Round(time.Duration(math.Pow10(k - n)))
+	}
+	return d
+}
+
+func digits(d time.Duration) int {
+	if d < 0 {
+		d = -d
+	}
+	i := 1
+	for d > 9 {
+		d /= 10
+		i++
+	}
+	return i
+}
+
+func formatLog(logFormatParts []string, param *gin.LogFormatterParams) string {
+	statusFormatted := fmt.Sprintf("%3d", param.StatusCode)
+
+	if param.IsOutputColor() {
```

Comment:
Let's also add the color back for the method. That was useful for quickly seeing request types as they came in. I think I also prefer having the background color as the default, but allowing for it to be changed would be nice too.